### DISCUSSION TITLE
Using Discord Auth Provider & Setting Custom Data

### DISCUSSION THREAD

**Discussion by gaurishhs** [nextauthjs/next-auth#4825] [Help]
### Question 💬

Hello! i'm using discord provider and setting the guilds of the user but they don't come when i use the `useSession` hook.

### How to reproduce ☕️

```
callbacks: {
        async signIn(params) {
            oauth.getUserGuilds(params.account.access_token as any).then(async (guilds) => {
                params.user.guilds = guilds;
            }).catch((e) => {
                return false
            })
            return true
        },
    }
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ivan-kleshnin** [comment]
This thread may help you: https://github.com/nextauthjs/next-auth/discussions/4538
